File: Project Android Studio/QuoteApp/app/src/main/java/com/komputerkit/quoteapp/MainActivity.java
package com.komputerkit.quoteapp;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private TextView textViewQuote;
    private TextView textViewAuthor;
    private Button buttonShare;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        
        // Initialize views
        textViewQuote = findViewById(R.id.textViewQuote);
        textViewAuthor = findViewById(R.id.textViewAuthor);
        buttonShare = findViewById(R.id.buttonShare);
        
        // Set click listener for share button
        buttonShare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                shareQuote();
            }
        });
        
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(android.R.id.content), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }
    
    private void shareQuote() {
        // Get the quote and author text
        String quote = textViewQuote.getText().toString();
        String author = textViewAuthor.getText().toString();
        
        // Combine quote and author
        String shareText = quote + "\n\n" + author;
        
        // Create implicit intent with ACTION_SEND
        Intent shareIntent = new Intent(Intent.ACTION_SEND);
        shareIntent.setType("text/plain");
        shareIntent.putExtra(Intent.EXTRA_TEXT, shareText);
        shareIntent.putExtra(Intent.EXTRA_SUBJECT, "Quote from Quote App");
        
        // Start chooser to let user select app to share with
        Intent chooser = Intent.createChooser(shareIntent, "Share Quote");
        if (shareIntent.resolveActivity(getPackageManager()) != null) {
            startActivity(chooser);
        }
    }
}

File: Project Android Studio/WavesofFoodAdmin/app/src/main/java/com/komputerkit/wavesoffoodadmin/MainActivity.kt
package com.komputerkit.wavesoffoodadmin

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.lifecycle.lifecycleScope
import com.komputerkit.wavesoffoodadmin.activity.LoginActivity
import com.komputerkit.wavesoffoodadmin.activity.MenuManagementActivity
import com.komputerkit.wavesoffoodadmin.activity.OrderManagementActivity
import com.komputerkit.wavesoffoodadmin.activity.UserManagementActivity
import com.komputerkit.wavesoffoodadmin.databinding.ActivityMainBinding
import com.komputerkit.wavesoffoodadmin.repository.OrderRepository
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    
    private lateinit var binding: ActivityMainBinding
    private val orderRepository = OrderRepository()
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        setupToolbar()
        setupWindowInsets()
        setupClickListeners()
        loadDashboardData()
        
        // Test Firestore connection
        testFirestoreConnection()
        
        // Check if admin is logged in
        checkAdminAuth()
    }
    
    private fun setupToolbar() {
        setSupportActionBar(binding.toolbar)
    }
    
    private fun setupWindowInsets() {
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }
    
    private fun setupClickListeners() {
        binding.cardMenuManagement.setOnClickListener {
            startActivity(Intent(this, MenuManagementActivity::class.java))
        }
        
        binding.cardOrderManagement.setOnClickListener {
            startActivity(Intent(this, OrderManagementActivity::class.java))
        }
        
        binding.cardUserManagement.setOnClickListener {
            startActivity(Intent(this, UserManagementActivity::class.java))
        }
    }
    
    private fun loadDashboardData() {
        lifecycleScope.launch {
            try {
                val stats = orderRepository.getOrderStats()
                
                binding.tvTotalOrders.text = stats["total"]?.toString() ?: "0"
                binding.tvPendingOrders.text = stats["pending"]?.toString() ?: "0"
                
                // Update welcome message with admin name
                val currentUser = FirebaseHelper.getCurrentUser()
                binding.tvWelcome.text = "Selamat Datang, ${currentUser?.email ?: "Admin"}"
                
            } catch (e: Exception) {
                Utils.showToast(this@MainActivity, "Gagal memuat data dashboard")
            }
        }
    }
    
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }
    
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_refresh -> {
                loadDashboardData()
                Utils.showToast(this, "Data diperbarui")
                true
            }
            R.id.action_logout -> {
                logout()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
    
    private fun logout() {
        FirebaseHelper.getAuth().signOut()
        startActivity(Intent(this, LoginActivity::class.java))
        finish()
    }
    
    override fun onResume() {
        super.onResume()
        loadDashboardData()
    }
    
    private fun testFirestoreConnection() {
        // Test dengan mengambil beberapa data dari collection foods
        FirebaseHelper.getFoodsCollection()
            .limit(1)
            .get()
            .addOnSuccessListener { documents ->
                if (!documents.isEmpty) {
                    Utils.showToast(this, "Firestore connected! Found ${documents.size()} food items")
                } else {
                    Utils.showToast(this, "Firestore connected but no food data found")
                }
            }
            .addOnFailureListener { exception ->
                Utils.showLongToast(this, "Firestore connection failed: ${exception.message}")
            }
    }
    
    private fun checkAdminAuth() {
        if (!FirebaseHelper.isUserLoggedIn()) {
            // Redirect to login activity
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
            return
        }
        
        FirebaseHelper.isCurrentUserAdmin { isAdmin ->
            if (!isAdmin) {
                Utils.showLongToast(this, "Access denied: Admin privileges required")
                // Sign out non-admin user
                FirebaseHelper.getAuth().signOut()
                startActivity(Intent(this, LoginActivity::class.java))
                finish()
            }
        }
    }
}